src/main/java/org/jenkinsci/plugins/github_branch_source/BranchDiscoveryTrait.java

```diff
@@ -239,6 +243,9 @@ public boolean isExcluded(@NonNull SCMSourceRequest request, @NonNull SCMHead he
                     if (headRepo != null // head repo can be null if the PR is from a repo that has been deleted
                             && p.getBase().getRepository().getFullName().equalsIgnoreCase(headRepo.getFullName())
                             && p.getHead().getRef().equals(head.getName())) {
+                        LOGGER.log(Level.WARNING, "Ignoring {} because "
```

Comment:
Warning is even higher priority than `INFO`.  Is that really needed here?

Reply:
No particular preference if there's some hard guideline. But by the idea and events that led to this PR, being that someone made a PR from release-candidate branch to master and we lost buildability of that branch due to policy - invisibly to everyone concerned so it needed a fair bit of log inspection and code interpolating and guesswork and eventually a look at configs, heightened visibility with a "WARNING, skipping what you might want to be used" makes sense.

Reply:
@jimklimov  
Yeah, logging it as a warning to the a listener local to the scan would be fine. I was considering it terms logging this in the overall system log as warning would get really noisy. 

Speaking of which, `request.listener().getLogger()` will get PrintStream you can write messages to that should show up in the scan log.  I think.

https://github.com/jenkinsci/scm-api-plugin/blob/master/src/main/java/jenkins/scm/api/trait/SCMSourceRequest.java#L377-L379

Reply:
Thanks for the idea, I did not see that method. At least syntactically (for building) that works well :)

Not sure I know how to unit-test this though.